src/AaveV2CollectorContractConsolidation.sol

```diff
@@ -0,0 +1,144 @@
+    address public constant ABUSD = 0xA361718326c15715591c299427c62086F69923D9;
+    address public constant BUSD = 0x4Fabb145d64652a948d72533023f6E7A623C7C53;
+    address public constant ZRX = 0xE41d2489571d322189246DaFA5ebDe1F4699F498;
+    address public constant AZRX = 0xDf7FF54aAcAcbFf42dfe29DD6144A69b629f8C9e;
+    address public constant ARENFIL = 0x514cd6756CCBe28772d4Cb81bC3156BA9d1744aa;
+    address public constant AENS = 0x9a14e23A58edf4EFDcB360f68cd1b95ce2081a2F;
+    address public constant ADPI = 0x6F634c6135D2EBD550000ac92F494F9CB8183dAe;
+
+    event Swap(address indexed tokenIn, address indexed tokenOut, uint256 amountIn, uint256 amountOut);
+
+    /// Not enough token left to swap
+    error NotEnoughTokens();
+    /// Oracle price is 0 or lower
+    error InvalidOracleAnswer();
+    /// Need to request more than 0 tokens out
+    error OnlyNonZeroAmount();
+    /// Token is not available for swap
+    error UnsupportedToken();
+
+    struct Asset {
+        uint256 quantity;
+        uint48 premium;
+        address oracle;
+        bool ethFeedOnly;
+    }
+
+    mapping(address => Asset) public assets;
+
+    constructor() {
+        assets[ARAI] = Asset(6740239e16, 100, 0x483d36F6a1d063d580c7a24F9A42B346f3a69fbb, false); // Custom Feed
+        assets[AAMPL] = Asset(15891248e7, 300, 0xe20CA8D7546932360e37E9D72c1a47334af57706, false); // Monitored Feed
+        assets[AFRAX] = Asset(2869022e16, 75, 0xB9E1E3A9feFf48998E45Fa90847ed4D467E8BcfD, false);
+        assets[FRAX] = Asset(1512479102e11, 75, 0xB9E1E3A9feFf48998E45Fa90847ed4D467E8BcfD, false);
+        assets[AENS] = Asset(7047e16, 300, 0x5C00128d4d1c2F4f652C267d7bcdD7aC99C16E16, false);
+        assets[SUSD] = Asset(9040e16, 75, 0x8e0b7e6062272B5eF4524250bFFF8e5Bd3497757, true); // Only ETH
+        assets[ASUSD] = Asset(1148320e16, 75, 0x8e0b7e6062272B5eF4524250bFFF8e5Bd3497757, true); // Only ETH
+        assets[TUSD] = Asset(160409e16, 75, 0xec746eCF986E2927Abd291a2A1716c940100f8Ba, false);
+        assets[ATUSD] = Asset(608004e16, 75, 0xec746eCF986E2927Abd291a2A1716c940100f8Ba, false);
+        assets[AMANA] = Asset(1622740e16, 200, 0x56a4857acbcfe3a66965c251628B1c9f1c408C19, false);
+        assets[MANA] = Asset(33110e16, 200, 0x56a4857acbcfe3a66965c251628B1c9f1c408C19, false);
+        assets[ADPI] = Asset(1359e16, 300, 0xD2A593BF7594aCE1faD597adb697b5645d5edDB2, false); 
+        assets[ABUSD] = Asset(364085e16, 75, 0x833D8Eb16D306ed1FbB5D7A2E019e106B960965A, false);
+        assets[BUSD] = Asset(3399064e14, 75, 0x833D8Eb16D306ed1FbB5D7A2E019e106B960965A, false);
+        assets[ZRX] = Asset(10719e16, 300, 0x2885d15b8Af22648b98B122b22FDF4D2a56c6023, false);
+        assets[AZRX] = Asset(877140e16, 300, 0x2885d15b8Af22648b98B122b22FDF4D2a56c6023, false);
+        assets[AUST] = Asset(89239797e4, 200, address(0), false); // NO FEED
+        assets[ARENFIL] = Asset(41067e16, 300, address(0), false); // NO FEED
```

Comment:
What're you doing in the case of No feed?

Reply:
Just saw that you're handling the case of address(0), in `getAmountIn()`. Why even have these assets here in the first place if they're not going to be supported?

Reply:
I'm actually waiting on BGD to see which Oracle to use, we need to support these tokens.